test/router/path_trie_tail_test.dart

```diff
@@ -0,0 +1,265 @@
+      expect(resSpecific, isNotNull);
+      expect(resSpecific!.value, 1);
+      expect(resSpecific.matched.path, '/foo/bar');
+      expect(resSpecific.remaining.path, '/baz/qux');
+
+      final resGeneral = trie.lookup(NormalizedPath('/foo/other/path'));
+      expect(resGeneral, isNotNull);
+      expect(resGeneral!.value, 2);
+      expect(resGeneral.matched.path, '/foo');
+      expect(resGeneral.remaining.path, '/other/path');
+    });
+
+    test(
+        'Given a trie with path /user/:id/files/**, '
+        'when /user/42/files/docs/report.pdf is looked up, '
+        'then it matches with correct parameter and remaining path', () {
+      trie.add(NormalizedPath('/user/:id/files/**'), 1);
+      final result =
+          trie.lookup(NormalizedPath('/user/42/files/docs/report.pdf'));
+      expect(result, isNotNull);
+      expect(result!.value, 1);
+      expect(result.parameters, equals({#id: '42'}));
+      expect(result.matched.path, '/user/42/files');
+      expect(result.remaining.path, '/docs/report.pdf');
+    });
+
+    test(
+        'Given a trie with path /data/export/**, '
+        'when /data (shorter than prefix) is looked up, '
+        'then no match is found', () {
+      trie.add(NormalizedPath('/data/export/**'), 1);
+      expect(trie.lookup(NormalizedPath('/data')), isNull);
+    });
+
+    test(
+        'Given an empty trie, '
+        'when adding a path like /**foo (tail not a full segment), '
+        'then an ArgumentError is thrown', () {
+      expect(() => trie.add(NormalizedPath('/downloads/**foo'), 1),
+          throwsArgumentError);
+    });
+
+    group('Tail and Other Segment interaction validation', () {
+      test(
+          'Given a trie with /test/**, '
+          'when adding /test/:id (parameter after tail at same level), '
+          'then an ArgumentError is thrown', () {
+        trie.add(NormalizedPath('/test/**'), 1);
+        expect(() => trie.add(NormalizedPath('/test/:id'), 2),
+            throwsArgumentError);
+      });
+
+      test(
+          'Given a trie with /test/:id, '
+          'when adding /test/** (tail after parameter at same level), '
+          'then an ArgumentError is thrown', () {
+        trie.add(NormalizedPath('/test/:id'), 1);
+        expect(
+            () => trie.add(NormalizedPath('/test/**'), 2), throwsArgumentError);
+      });
+
+      test(
+          'Given a trie with /test/**, '
+          'when adding /test/* (wildcard after tail at same level), '
+          'then an ArgumentError is thrown', () {
+        trie.add(NormalizedPath('/test/**'), 1);
+        expect(
+            () => trie.add(NormalizedPath('/test/*'), 2), throwsArgumentError);
+      });
+
+      test(
+          'Given a trie with /test/*, '
+          'when adding /test/** (tail after wildcard at same level), '
+          'then an ArgumentError is thrown', () {
+        trie.add(NormalizedPath('/test/*'), 1);
```

Comment:
Q: Was this "your choice" or an implementation detail?

I suppose one could still argue that `/*` is more precise for `/foo` than `/**`, but either way, I don't see this relevant in practice. Just wondered whether that would make the internal implementation more complicated (for little gain), or is just a safeguard against confusing usage.

Reply:
To elaborate with an example

Consider:
1. /*/foo
2. /**

Lookup /bar/nofoo/